### PR TITLE
feat: expand testing suite

### PR DESCRIPTION
**Pull request by kjanat**
This pull request enhances the Bun testing environment setup and improves mutation-related test coverage. The main changes include expanding the DOM and storage mocks for tests, removing global canvas mocking in favor of local, and significantly extending mutation tool and prediction test cases to cover more edge conditions and behaviors.

**Test Environment Improvements:**

* Expanded the global DOM setup in `bun-test-setup.ts` to include more HTML elements and event types, and added isolated `localStorage` and `sessionStorage` mocks for better test isolation. [[1]](diffhunk://#diff-eb9ad9d16e763cc597e7732255c898f8edd6fdae9afc1279510427214057cc68L3-R49) [[2]](diffhunk://#diff-eb9ad9d16e763cc597e7732255c898f8edd6fdae9afc1279510427214057cc68L36-R84)
* Removed the global mocking of `HTMLCanvasElement.getContext` and related canvas methods from the test setup. Tests now mock canvas context locally where needed.

**Test Coverage Enhancements:**

* Added local canvas mocking utilities (`mockCanvasContext`, `restoreCanvasContext`) and updated all relevant test files (`educational-validation.test.ts`, `mutation-predictor.test.ts`, `performance-benchmarks.test.ts`) to use these for improved isolation and reliability of rendering-dependent tests. [[1]](diffhunk://#diff-a2dd5698de6678b9b61e7bc816936a35852ac25aac35796796361c4ebb1b9d04L3-R8) [[2]](diffhunk://#diff-a2dd5698de6678b9b61e7bc816936a35852ac25aac35796796361c4ebb1b9d04R32-R35) [[3]](diffhunk://#diff-a24cc68ef477b30c7004a343d7bbefdf54ccd04bf4666f238fc7958d08b95a66L6-R10) [[4]](diffhunk://#diff-a24cc68ef477b30c7004a343d7bbefdf54ccd04bf4666f238fc7958d08b95a66R26-R29) [[5]](diffhunk://#diff-004e425ce7197406b4d662b2cc492dd4ce37ddd150ceccb45ff5cd8b1d90526fL3-R9) [[6]](diffhunk://#diff-004e425ce7197406b4d662b2cc492dd4ce37ddd150ceccb45ff5cd8b1d90526fR26-R33)

**Mutation Tool Test Extensions:**

* Added new edge case and boundary tests for all mutation functions in `mutations.test.ts`, including checks for invalid positions, correct handling at genome boundaries, and specific behaviors for synonymous, missense, nonsense, point, insertion, and deletion mutations. [[1]](diffhunk://#diff-9a47e792ac53f824e4b06e3447a17d8e1039e9a9090ad3346272dc8f260b48aaR55-R72) [[2]](diffhunk://#diff-9a47e792ac53f824e4b06e3447a17d8e1039e9a9090ad3346272dc8f260b48aaR109-R118) [[3]](diffhunk://#diff-9a47e792ac53f824e4b06e3447a17d8e1039e9a9090ad3346272dc8f260b48aaR144-R159) [[4]](diffhunk://#diff-9a47e792ac53f824e4b06e3447a17d8e1039e9a9090ad3346272dc8f260b48aaR193-R209) [[5]](diffhunk://#diff-9a47e792ac53f824e4b06e3447a17d8e1039e9a9090ad3346272dc8f260b48aaR237-R253) [[6]](diffhunk://#diff-9a47e792ac53f824e4b06e3447a17d8e1039e9a9090ad3346272dc8f260b48aaR275-R292)

**Mutation Prediction Test Extensions:**

* Significantly expanded the mutation impact predictor tests to validate classification for LOCAL, MAJOR, and CATASTROPHIC impacts, descriptions, confidence levels, color/position analysis, and edge cases for truncation and frameshift scenarios.

## Summary by CodeRabbit

* **Tests**
  * Major expansion: dozens of test files upgraded from placeholders to comprehensive, runnable suites covering UI, export, rendering, analytics, and edge cases.

* **Chores**
  * CI/CD and quality checks added and improved (typecheck, linting, duplication, workflows).
  * Demo pages and playground moved to modular assets (external scripts + styles).
  * Design system introduced with centralized CSS variables and many new stylesheets.

* **New Features**
  * New teacher export/download utility for student data.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by kjanat** [comment]
@coderabbitai resume

File: .github/workflows/super-linter.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          # To report GitHub Actions status checks
+          GITHUB_TOKEN: ${{ github.token }}
+          IGNORE_GITIGNORED_FILES: true
+          JSCPD_CONFIG_FILE: package.json
+          CHECKOV_FILE_NAME: .github/linters/checkov.yaml
+          VALIDATE_BIOME_FORMAT: false
+          VALIDATE_BIOME_LINT: false
+
+      - name: Prevents job failure
+        run: echo "Hi!"
```
[the remark this answers is not in the document]
Comment (kjanat) [reply .github/workflows/super-linter.yml]: As the job name says, it prevents the job from failing, as continue-on-error is true in the step before...
